### PR TITLE
Add `sails-js`

### PR DESCRIPTION
**Pull request by wentelteefje**
This PR adds documentation for `sails-js` to the Wiki while also re-arranging some pages to increase UX.

### CODE REVIEW

File: docs/sails-js/sails-js.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+---
+sidebar_position: 1
+sidebar_label: Sails-JS
+---
+
+# Sails-JS
+
+The `sails-js` library is a powerful tool designed to help developers interact with their Sails applications on Vara Network. It facilitates seamless communication between your TypeScript projects and on-chain programs by providing two primary methods for generating client libraries from a Sails IDL (Interface Definition Language):
```
Comment (AndrePanin) [inline docs/sails-js/sails-js.md:8]: At the next chapter - "The Gear-JS API offers a set of utilities, libraries, and tools that enable JavaScript applications to interact with programs running on the Vara network ..."
Could you pls add some inputs at the beginning about what's the difference between sails-js and gear-js? when and why dapp devs should choose one or another.

**Review by AndrePanin — APPROVED** [review]
Awesome! Just updated chapters ordering.

File: docs/sails-js/client-generation.md [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+sidebar_position: 3
+sidebar_label: Client Generation
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Client Generation
+
+The `sails-js-cli` is a command-line tool designed to generate TypeScript client libraries from Sails IDL files. It automates the creation of fully functional client libraries based on the interfaces defined in the Sails framework, streamlining development and ensuring consistency between client and on-chain programs.
```
Comment (nikitayutanov) [inline docs/sails-js/client-generation.md:11]: I think it's worth to attach a link to the `Overview` section with the specifics of the cli over raw sails.

**Comment by nikitayutanov** [comment]
What about the Sails functionality in `@gear-js/react-hooks`? Current [README](https://github.com/gear-tech/gear-js/tree/main/utils/gear-hooks#sails) briefly covers it, but I believe frontend devs would benefit from more detailed instructions.

Like Dmitry said, it's CLI generation is not ready yet, but devs can still utilize the hooks manually.

Since the core functionality is based on the generated `lib.ts`, there's no need to repeat information mentioned in the `Overview` and `sails-js-cli` sections. We can just emphasize the slight adjustments that these hooks offer for specific React and general frontend workflows.

File: docs/sails-js/client-generation.md [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+
+Event subscription methods allow subscribing to specific events emitted by the program.
+
+```javascript
+program.subscribeToSomeEvent((data) => {
+  console.log(data);
+});
+```
+
+## React Hooks
```
[the remark this answers is not in the document]
Comment (wentelteefje) [reply docs/sails-js/client-generation.md]: I commented out the respective lines for now

File: docs/sails-js/react-hooks.md [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+    </main>
+  );
+}
+
+export { App };
+```
+
+## Prepare Hooks
+
+When performing long-running asynchronous work in response to a user interaction (e.g., using "transactional" hooks such as [`useSendProgramTransaction`](#usesendprogramtransaction)), the user only receives feedback on the transaction status after invoking the click handler. Prepare hooks can be used to eagerly obtain values before user interaction or to perform upfront validation before sending a transaction.
```
Comment (nikitayutanov) [inline docs/sails-js/react-hooks.md:167]:
```suggestion
When performing long-running asynchronous work in response to a user transaction using [`useSendProgramTransaction`](#usesendprogramtransaction)) hook, the user only receives feedback on the transaction status after transaction is complete. Prepare hooks can be used to eagerly obtain values before user interaction or to perform upfront validation before sending a transaction.
```

I guess, we can be specific here. As of right now there's only one purpose.

**Review by nikitayutanov — APPROVED** [review]

**Pull request merged**